src/wallet/test/coinselector_tests.cpp

```diff
@@ -548,19 +542,20 @@ BOOST_AUTO_TEST_CASE(knapsack_solver_test)
 
           // Again, we only create the wallet once to save time, but we still run the coin selection RUN_TESTS times.
           for (int i = 0; i < RUN_TESTS; i++) {
-            // picking 50 from 100 coins doesn't depend on the shuffle,
+            // picking 50.5 from 100 coins doesn't depend on the shuffle,
             // but does depend on randomness in the stochastic approximation code
-            BOOST_CHECK(testWallet.SelectCoinsMinConf(50 * COIN, filter_standard, vCoins, setCoinsRet , nValueRet, coin_selection_params, bnb_used));
-            BOOST_CHECK(testWallet.SelectCoinsMinConf(50 * COIN, filter_standard, vCoins, setCoinsRet2, nValueRet, coin_selection_params, bnb_used));
+            // Use 50.5 to be sure that BnB doesn't match
```

Comment:
In a493cbcb26: it seems a little hacky to use 50.5 to make it use the knapsack solver, why not just call `KnapsackSolver` directly since that's what you're testing?

Reply:
Good idea, done.